src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1982,7 +1982,7 @@ private bool ScanIdentifier_CrefSlowPath(ref TokenInfo info)
                             else if (_identLen > 0 && consumedChar > 127 && SyntaxFacts.IsIdentifierPartCharacter(consumedChar))
                             {
                                 //// BUG 424819 : Handle identifier chars > 0xFFFF via surrogate pairs
-                                if (SyntaxFacts.IsFormattingChar(consumedChar))
+                                if (UnicodeCharacterUtilities.IsFormattingChar(consumedChar))
```

Comment:
There are many calls to SyntaxFacts character classification methods in Lexer.cs. They should be rewritten to directly use the versions in UnicodeCharacterUtilities if possible, as the lexer is in the inner loop of the compiler.